### PR TITLE
Cherry picking DTS bug fixes to U3

### PR DESCRIPTION
**Pull request by ds-ms**

### CODE REVIEW

File: Tasks/ServiceFabricDeploy/Tests/CreateDiffPkg.ps1 [hunk 1]
```diff
@@ -142,13 +148,17 @@ Microsoft.PowerShell.Core\Import-Module "$PSScriptRoot\..\Create-DiffPackage.psm
...
+Assert-WasCalled Copy-Item -LiteralPath $appManifestPath $appManifestDiffPath -Force
+Assert-WasCalled Copy-Item -LiteralPath $serviceManifestPath1 $serviceManifestDiffPath1 -Force -Times 0
+Assert-WasCalled Copy-Item -LiteralPath $serviceManifestPath2 $serviceManifestDiffPath2 -Force
+Assert-WasCalled Copy-Item -LiteralPath $serviceManifestPath3 $serviceManifestDiffPath3 -Force -Times 0
+Assert-WasCalled Copy-Item -LiteralPath $serviceManifestPath4 $serviceManifestDiffPath4 -Force
+Assert-WasCalled Copy-Item -LiteralPath $codePkg1 $codeDiffPkg1 -Recurse -Times 0
+Assert-WasCalled Copy-Item -LiteralPath $codePkg2 $codeDiffPkg2 -Recurse
+Assert-WasCalled Copy-Item -LiteralPath $codeZippedPkg3 $codeZippedDiffPkg3 -Recurse -Times 0
+Assert-WasCalled Copy-Item -LiteralPath $codeZippedPkg4 $codeZippedDiffPkg4 -Recurse
+Assert-WasCalled Copy-Item -LiteralPath $configZippedPkg4 $configZippedDiffPkg4 -Recurse -Times 0
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricDeploy/Tests/CreateDiffPkg.ps1]: You might want to remove these additional lines. They do not exist any longer.

File: Tasks/ServiceFabricDeploy/Tests/CreateDiffPkg.ps1 [hunk 2]
```diff
@@ -131,9 +131,15 @@ Register-Mock Get-ServiceFabricServiceManifest {$serviceManifest2} -- -Applicati
 Register-Mock Get-ServiceFabricServiceManifest {$serviceManifest3} -- -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $applicationTypeVersion -ServiceManifestName "Stateless3Pkg"
 Register-Mock Get-ServiceFabricServiceManifest {$serviceManifest4} -- -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $applicationTypeVersion -ServiceManifestName "Stateless4Pkg"
 
-Register-Mock Copy-Item {} $appManifestPath $appManifestDiffPath -Force
-Register-Mock Test-Path { $true } -Path $codePkg3
-Register-Mock Test-Path { $true } -Path $codePkg4
+Register-Mock Copy-Item {} -LiteralPath $appManifestPath $appManifestDiffPath -Force
+Register-Mock Test-Path { $true } -- -LiteralPath $codePkg1
+Register-Mock Test-Path { $true } -- -LiteralPath $codePkg2
+Register-Mock Test-Path { $true } -- -LiteralPath $codeZippedPkg3
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricDeploy/Tests/CreateDiffPkg.ps1]: You might want to remove these additional lines. They do not exist any longer.

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts [hunk 3]
```diff
@@ -299,6 +299,7 @@ export class ResourceGroup {
                     let contents: string = "";
                     try {
                         contents = await response.readBody();
+                        contents = contents.replace(/^\uFEFF/, ''); // Remove UTF-8 BOM if present.
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: version update required for this?

Comment (ds-ms) [reply Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: done

**Pull request merged**